test-crawler/extractor/extractor.go

```diff
@@ -103,17 +128,17 @@ func parseContent(content string, treeCursor *sitter.TreeCursor, filePath string
 	var annotationParser a.Parser
 
 	fileData.Metadata = getMetadata(content, treeCursor, &annotationParser)
-	functions := getFunctionNodes(content, treeCursor, &annotationParser)
+	funcNodes := getFunctionNodes(content, treeCursor, &annotationParser)
 
-	for _, function := range functions {
+	for _, function := range funcNodes {
```

Comment:
Rename `function` to `funcNode`. See line 141: 
`function.Function.Name` would then be `funcNode.Function.Name`. 

Which is much clearer, no?